src/Services/PractitionerService.php

```diff
@@ -0,0 +1,265 @@
+     * payload.
+     */
+    public function getOne($uuid)
+    {
+        $processingResult = new ProcessingResult();
+
+        $isValid = $this->practitionerValidator->validateId("uuid", "users", $uuid, true);
+
+        if ($isValid !== true) {
+            $validationMessages = [
+                'uuid' => ["invalid or nonexisting value" => " value " . $uuid]
+            ];
+            $processingResult->setValidationMessages($validationMessages);
+            return $processingResult;
+        }
+
+        $sql = "SELECT  id,
+                        uuid,
+                        users.title as title,
+                        fname,
+                        lname,
+                        mname,
+                        federaltaxid,
+                        federaldrugid,
+                        upin,
+                        facility_id,
+                        facility,
+                        npi,
+                        email,
+                        active,
+                        specialty,
+                        billname,
+                        url,
+                        assistant,
+                        organization,
+                        valedictory,
+                        street,
+                        streetb,
+                        city,
+                        state,
+                        zip,
+                        phone,
+                        fax,
+                        phonew1,
+                        phonecell,
+                        users.notes,
+                        state_license_number,
+                        abook.title as abook_title,
+                        physician.title as physician_title,
+                        physician.codes as physician_code
+                FROM  users
+                LEFT JOIN list_options as abook ON abook.option_id = users.abook_type
+                LEFT JOIN list_options as physician ON physician.option_id = users.physician_type
+                WHERE uuid = ? AND npi is not null";
+
+        $uuidBinary = UuidRegistry::uuidToBytes($uuid);
+        $sqlResult = sqlQuery($sql, [$uuidBinary]);
+
+        $sqlResult['uuid'] = UuidRegistry::uuidToString($sqlResult['uuid']);
+        $processingResult->addData($sqlResult);
+        return $processingResult;
+    }
+
+
+    /**
+     * Inserts a new practitioner record.
+     *
+     * @param $data The practitioner fields (array) to insert.
+     * @return ProcessingResult which contains validation messages, internal error messages, and the data
+     * payload.
+     */
+    public function insert($data)
+    {
+        $processingResult = $this->practitionerValidator->validate(
+            $data,
+            PractitionerValidator::DATABASE_INSERT_CONTEXT
+        );
+
+        if (!$processingResult->isValid()) {
+            return $processingResult;
+        }
+
+        $freshPid = $this->getFreshId("id", "users");
+        $data['id'] = $freshPid;
```

Comment:
Note that id in users table is autoincrementing, so no reason to create an id here (doing this opens door to possible concurrency issues). Can just let database create the id naturally on the insert.

Reply:
Agreed :+1:  

Thanks :)